### PR TITLE
Upgrade and refactor swagger file to OpenApi 3.0 with param objs

### PR DESCRIPTION
**Pull request by rebeksburnett**

### CODE REVIEW

**Comment by rebeksburnett** [comment]
There are two big changes in this PR:  
 - automatic upgrade from 2.0 to 3.0. (This did a lot of strange changes to the quotations, and I couldn't find any real rules to follow there, so I just left it exactly as the tool created.)
 - change to use parameter objects (like done in the recurring-api docs)

I've used parameter objects now, but notice that there were places where a default value was also specified. I did not find any way to reuse the parameter object and add the default value, so I made a second parameter object there (see Authorization_with_default). I wondered if you had any thoughts about how to do that?

Also, the definitions for the same things (e.g., authorization) were all slightly different from path to path, so I tried to combine them into one common definition. I also copied some text from the recurring-api swagger file.

File: docs/swagger.yaml [hunk 1]
```diff
@@ -93,58 +80,41 @@ paths:
...
-          type: string
-          default: C948DFD1546347568874C4DDC93A2E3C
-        - name: User-Token
-          in: header
-          description: JWT-TOKEN
-          required: true
-          type: string
-          default: eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJvcmdhbml6YXRpb25OdW1iZXIiOiI5MTg3MTM4NjciLCJpc3MiOiJ2aXBwcy5ubyIsImV4cCI6MTY0NDc0Mzg5NywiYWNjb3VudE51bWJlciI6IjE4MDAwMDAwMDAwIn0.mqO4RsidM_-twRfCvgNnXUUPrXaTgFVMvkJTVmJgtmbHziMVxHnG9moHpEj90_A3MRPp_jucdgMnqixgExOykJCLX6VsiUB0qMLXTgBE7Mmkz3lk0jCgv7OwYDJuLksrXvLepD68KdyDTCVwPxdgqh_H9Tea7S3gSYUnYoNkNP5h7Eyvai6U7BuJVcoE5ckO1ZjGmOvNlfHUbAZYBH8I5RV96bnmEtdtkojrkoZtT7HjueyTZODfdaOyvI5VfMLKbr9qGkYKP0UBKTp9ST5eV-W_Y062PgOsfRNXL2-ltKgn0vDkpUWYGiA8Y98ondXLF1kQIc8pAFPAwhzWG31oM_wKeOjjjK-2lMg1X1LjTk-zbq0HBilhQzbXdW0PHws0p1UZgviW-bdIwRfcjn1ysEqzS0HzDuaRTGIQDeld3MrtcwLLce7PT4rSGZeoJM-pUTidztzX48pFUixLm-aa1-SWnbRIqUufmsylSJNZXjgS06pISGUMKPiAdSjINY17x8YkeptS1B_r56oshRY8BFffCPxtMJI1BAWYQBCQ8WvUadlWNrQ477kUuvauEPRK_eyXT83cANaEa39lBPz4J3PHAE6grV4c3G_rClFzaFl1mzrhUABKJ3CpP_q7LmkboXSI5xx0BkyIb4wG_eG1s7_E_P_T0hIj3CTBNZnu7mI
+      
+        - $ref: "#/components/parameters/Authorization-with-default"
```
Comment (tht13) [inline docs/swagger.yaml:85]: I think we can make all authorisations with default

File: docs/swagger.yaml [hunk 2]
```diff
@@ -93,58 +80,41 @@ paths:
...
+        - $ref: "#/components/parameters/Merchant-Serial-Number"
+        - $ref: "#/components/parameters/Ocp-Apim-Subscription-Key"
+        - $ref: "#/components/parameters/Psp-Id-with-default"
```
Comment (tht13) [inline docs/swagger.yaml:88]: same here

File: docs/swagger.yaml [hunk 3]
```diff
@@ -154,759 +124,870 @@ paths:
...
-          description: makePaymentRequest
+          description: Token provided by the PSP as 'makePaymentToken' in the
+
+            `/v3/psppayments/init` call.
+
           required: true
           schema:
-            $ref: '#/definitions/MakePaymentRequest'
+            type: string
+        - $ref: "#/components/parameters/Vipps-System-Name"
```
Comment (tht13) [inline docs/swagger.yaml:208]: this is an outbound request, no plugin headers here

File: docs/swagger.yaml [hunk 4]
```diff
@@ -154,759 +124,870 @@ paths:
...
+        required: true
       responses:
-        '200':
+        "200":
           description: OK
-          schema:
-            $ref: '#/definitions/Response'
-        '400':
+          content:
+            "*/*":
```
Comment (tht13) [inline docs/swagger.yaml:265]: should be application/json

File: docs/swagger.yaml [hunk 5]
```diff
@@ -154,759 +124,870 @@ paths:
...
+      description: >-
+        Bearer JWT-TOKEN. 
+            The access token is a base64-encoded string that is required for all API calls.
+            It is a JWT (JSON Web Token).
+            The access token is fetched from the `POST:/accesstoken/get` endpoint.
+            It is valid for 1 hour in the test environment and 24 hours in the production environment.
+      required: true
+      schema:
+        type: string
+    Authorization-with-default:
```
Comment (tht13) [inline docs/swagger.yaml:391]: combine

File: docs/swagger.yaml [hunk 6]
```diff
@@ -154,759 +124,870 @@ paths:
...
+      in: header
+      name: Psp-Id
+      description: >-
+        PSP id is provided by Vipps. This is the PSP's id, and the same for
+            all the PSP's merchants. GUID value for PSPs created after start of 2021.
+      required: true
+      schema:
+        type: string
+        default: C948DFD1546347568874C4DDC93A2E3C
+      example: C948DFD1546347568874C4DDC93A2E3C
```
Comment (tht13) [inline docs/swagger.yaml:446]: combine

File: docs/swagger.yaml (hunk 5 above)
Comment (rebeksburnett) [reply docs/swagger.yaml:391]: Thanks

File: docs/swagger.yaml (hunk 1 above)
Comment (rebeksburnett) [reply docs/swagger.yaml:85]: @tht13 Can you look at the Psp-Id-with-default, too?

Comment (tht13) [reply docs/swagger.yaml:85]: merge them

File: docs/swagger.yaml [hunk 7]
```diff
@@ -154,759 +124,845 @@ paths:
...
-        '500':
+        "500":
           description: Internal Server Error
-          schema:
-            type: array
-            items:
-              $ref: '#/definitions/ErrorDetails'
-        '503':
+          content:
+            "*/*":
```
Comment (tht13) [inline docs/swagger.yaml:278]: all */* should be fixed

Comment (rebeksburnett) [reply docs/swagger.yaml:278]: Do you mean that everywhere it says "*/*", it should be      application/json       ?

Also, I have see it as          application/json;charset=UTF-8    

Should they all include charset=UTF-8?

Comment (rebeksburnett) [reply docs/swagger.yaml:278]: Alf clarified it for me.  Update all of them to application/json. Leave off charset=UTF-8.

**Review by tht13 — APPROVED** [review]

**Comment by tht13** [comment]
LGTM 🚀

**Pull request merged**